### PR TITLE
Adding RAC result extensions

### PR DESCRIPTION
**Pull request by danielaRiesgo**
## Summary ##

Adding RAC result extensions needed in Wolmo-Auth.
They transform the value to a result value for values and errors and filtering.

### CODE REVIEW

File: Core/Extensions/ReactiveCocoa/Signal.swift [hunk 1]
```diff
@@ -7,18 +7,77 @@
...
+    
+    /**
+        Transforms a `Signal<ResultProtocol<Value2, Error2>, Error>` to `Signal<Value2, Error>`,
+        ignoring all `Error2` events.
+     
+        It may be considered similar to the `values` signal of an `Action`.
+    */
+    func filterValues() -> Signal<Value.Value, Error> {
+        return filter {
+            if let _ = $0.value {
```
Comment (pablog) [inline Core/Extensions/ReactiveCocoa/Signal.swift]: Why not `return $0.value != .none`?

File: Core/Extensions/ReactiveCocoa/Signal.swift [hunk 2]
```diff
@@ -7,18 +7,77 @@
...
+    
+    /**
+         Transforms a `Signal<ResultProtocol<Value2, Error2>, Error>` to `Signal<Error2, Error>`,
+         ignoring all `Value2` events.
+     
+         It may be considered similar to the `errors` signal of an `Action`.
+     */
+    func filterErrors() -> Signal<Value.Error, Error> {
+        return filter {
+            if let _ = $0.error {
```
Comment (pablog) [inline Core/Extensions/ReactiveCocoa/Signal.swift]: Same

File: Core/Extensions/ReactiveCocoa/SignalProducer.swift [hunk 3]
```diff
@@ -21,4 +22,65 @@ public extension SignalProducerProtocol {
...
+    /**
+         Transforms a `SignalProducer<ResultProtocol<Value2, Error2>, Error>`
+         to `SignalProducer<Value2, Error>`, ignoring all `Error2` events.
+         
+         It may be considered similar to the `values` signal of an `Action`,
+         but for producers.
+     */
+    func filterValues() -> SignalProducer<Value.Value, Error> {
+        return filter {
+            if let _ = $0.value {
```
Comment (pablog) [inline Core/Extensions/ReactiveCocoa/SignalProducer.swift]: Same

File: Core/Extensions/ReactiveCocoa/SignalProducer.swift [hunk 4]
```diff
@@ -21,4 +22,65 @@ public extension SignalProducerProtocol {
...
+    /**
+         Transforms a `SignalProducer<ResultProtocol<Value2, Error2>, Error>`
+         to `SignalProducer<Error2, Error>`, ignoring all `Value2` events.
+         
+         It may be considered similar to the `errors` signal of an `Action`,
+         but for producers.
+     */
+    func filterErrors() -> SignalProducer<Value.Error, Error> {
+        return filter {
+            if let _ = $0.error {
```
Comment (pablog) [inline Core/Extensions/ReactiveCocoa/SignalProducer.swift]: Same

**Comment by woloxci** [comment]
:+1: There are no linter violations.

File: Core/Extensions/ReactiveCocoa/Signal.swift (hunk 1 above)
Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/Signal.swift]: I don't know why it doesn't work, it doesn't let me compare them.

Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/Signal.swift]: It says that I must add a `as! _?` which is super weird, but even if I do, I get an error `expected type after as`.

File: CoreTests/Extensions/ReactiveCocoa/SignalProducerSpec.swift [hunk 5]
```diff
@@ -35,28 +34,155 @@ public class SignalProducerSpec: QuickSpec {
...
+                        expect($0.count).to(equal(2))
+                        done()
+                    }
+                }}
+                
+            }
+            
+        }
+        
+        describe("#filterValues") {
```
Comment (damfinkel) [inline CoreTests/Extensions/ReactiveCocoa/SignalProducerSpec.swift]: where is filterValues used here? Same for filterErrors

Comment (danielaRiesgo) [inline CoreTests/Extensions/ReactiveCocoa/SignalProducerSpec.swift]: <insert facepalm gif here>

**Pull request merged**